lib/features/children/add_member/pages/family_member_form_page.dart

```diff
@@ -0,0 +1,120 @@
+import 'package:givt_app/shared/widgets/buttons/givt_elevated_secondary_button.dart';
+
+class FamilyMemberFormPage extends StatefulWidget {
+  const FamilyMemberFormPage(
+      {required this.index, required this.totalCount, super.key});
+  final int index;
+  final int totalCount;
+  @override
+  State<FamilyMemberFormPage> createState() => _FamilyMemberFormPageState();
+}
+
+class _FamilyMemberFormPageState extends State<FamilyMemberFormPage> {
+  final _formKey = GlobalKey<FormState>();
+  final _nameController = TextEditingController();
+  final _emailController = TextEditingController();
+  final _ageController = TextEditingController();
+  List<bool> selections = [true, false];
+  int _amount = 5;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  void pop({bool isChildSelected = false}) {
+    if (_formKey.currentState!.validate()) {
+      Navigator.of(context).pop(
+        isChildSelected
+            ? Member(
+                firstName: _nameController.text,
+                age: int.parse(_ageController.text),
+                allowance: _amount,
+                type: ProfileType.Child,
+              )
+            : Member(
+                firstName: _nameController.text,
+                email: _emailController.text,
+                type: ProfileType.Parent,
+              ),
+      );
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final isLast = widget.index == widget.totalCount;
+    final isChildSelected = selections[0];
+    return Scaffold(
+      appBar: const GenerosityAppBar(
+        title: 'Set up Family',
+        leading: GenerosityBackButton(),
+      ),
+      body: SafeArea(
+        child: Center(
+          child: Padding(
+            padding: const EdgeInsets.all(24),
+            child: Column(
+              children: [
+                SmileyCounter(
+                  totalCount: widget.totalCount,
+                  index: widget.index,
+                ),
+                ChildOrParentSelector(
+                  selections: selections,
+                  onPressed: (int index) {
+                    setState(() {
+                      selections = [index == 0, index == 1];
+                    });
+                  },
+                ),
+                const SizedBox(height: 16),
+                FamilyMemberForm(
+                  formKey: _formKey,
+                  nameController: _nameController,
+                  emailController: _emailController,
+                  ageController: _ageController,
+                  allowanceAmount: _amount,
+                  onAmountChanged: (amount) {
+                    setState(() {
+                      _amount = amount;
+                    });
+                  },
+                  isChildSelected: isChildSelected,
+                ),
+                const SizedBox(height: 40),
+                const Spacer(),
+                if (isLast)
+                  GivtElevatedButton(
```

Comment:
Not per se for this PR but I forgot to think about how we would re-use this when you navigate here from the manage family screen. Because then showing these buttons is not based on first or last. But right now an idea doesn't fully come to mind yet 🤔